File: src/main/java/com/opdapp/model/DrugPackage.java
package com.opdapp.model;

import javax.persistence.*;

@Entity
public class DrugPackage {

    @Id
    @GeneratedValue
    private long drugPackageId;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof DrugPackage)) return false;

        DrugPackage that = (DrugPackage) o;

        return getDrugPackageId() == that.getDrugPackageId();
    }

    @Override
    public int hashCode() {
        return (int)getDrugPackageId();
    }

    @ManyToOne
    @JoinColumn(name = "drugId")

    private Drug drug;

    @ManyToOne
    @JoinColumn(name = "strengthId")
    private Strength strength;
    private double quantity;
    private double minOrderLevel;
    private double unitPrice;

    public long getDrugPackageId() {
        return drugPackageId;
    }

    public void setDrugPackageId(long drugPackageId) {
        this.drugPackageId = drugPackageId;
    }

    public Drug getDrug() {
        return drug;
    }

    public void setDrug(Drug drug) {
        this.drug = drug;
    }

    public Strength getStrength() {
        return strength;
    }

    public void setStrength(Strength strength) {
        this.strength = strength;
    }

    public double getQuantity() {
        return quantity;
    }

    public void setQuantity(double quantity) {
        this.quantity = quantity;
    }

    public double getMinOrderLevel() {
        return minOrderLevel;
    }

    public void setMinOrderLevel(double minOrderLevel) {
        this.minOrderLevel = minOrderLevel;
    }

    public double getUnitPrice() {
        return unitPrice;
    }

    public void setUnitPrice(double unitPrice) {
        this.unitPrice = unitPrice;
    }
}

File: frontend/src/app/drugcomponent/drugpackage.ts
import {Drug} from "./drug";
import {Strength} from "./strength";

export class Drugpackage {
  private _drugPackageId;
  private _drug: Drug;
  private _strength: Strength;
  private _quantity;


  get drugPackageId() {
    return this._drugPackageId;
  }

  set drugPackageId(value) {
    this._drugPackageId = value;
  }

  get drug(): Drug {
    return this._drug;
  }

  set drug(value: Drug) {
    this._drug = value;
  }

  get strength(): Strength {
    return this._strength;
  }

  set strength(value: Strength) {
    this._strength = value;
  }

  get quantity() {
    return this._quantity;
  }

  set quantity(value) {
    this._quantity = value;
  }
}
